clouds/bigquery/modules/sql/quadbin/QUADBIN_TOCHILDREN.sql

```diff
@@ -6,17 +6,34 @@ CREATE OR REPLACE FUNCTION `@@BQ_DATASET@@.QUADBIN_TOCHILDREN`
 (quadbin INT64, resolution INT64)
 RETURNS ARRAY<INT64>
 AS ((
-    IF(resolution < 0 OR resolution > 26 OR resolution < ((quadbin >> 52) & 0x1F),
+    IF(resolution < 0 OR resolution > 26 OR resolution < ((resolution >> 52) & 0x1F),
         ERROR('Invalid resolution'),
         (
-            WITH __zxy AS (
-                SELECT `@@BQ_DATASET@@.QUADBIN_TOZXY`(quadbin) AS tile
+            WITH
+            __constants AS (
+              SELECT
+                ~(0x1F << 52) AS zoom_level_mask,
+                          (1 << ((resolution - ((quadbin >> 52) & 0x1F)) << 1)) AS block_range,
+                CAST(SQRT((1 << ((resolution - ((quadbin >> 52) & 0x1F)) << 1))) as int) AS sqrt_block_range,
```

Comment:
Take into account SQRT is not required since we can do it in binary:

```sql
(1 << resolution_diff) AS sqrt_block_range,
```

Reply:
gosh... you are right!